lib/bamboo/email.ex

```diff
@@ -203,7 +203,27 @@ defmodule Bamboo.Email do
   end
 
   @doc ~S"""
-  Adds an attachment to the email
+  Adds an data attachment to the email
+
+  ## Example
+    put_attachment(email, %Bamboo.Attachment{})
+
+  Requires the fields filename and data of the %Bamboo.Attachment{} struct to be set.
+
+  ## Example
+    def create(conn, params) do
+      #...
+      email
+      |> put_attachment(%Bamboo.Attachment{filname: "event.ics", data: "BEGIN:VCALENDAR..."})
+      #...
+    end
+  """
+  def put_attachment(%__MODULE__{attachments: attachments} = email, %Attachment{filename: _filename, data: _data} = attachment) do
```

Comment:
Maybe we have 2 functions. One for the happy path and another for attachments that are missing data and filename

```elixir
def put_attachment(_email, %Attachment{filename: nil, data: nil} = attachment) do
  raise "You must pass a valid attachment, instead got: #{inspect attachment}"
end

def put_attachment(%__MODULE__{attachments: attachments} = email, %Attachment{} = attachment) do
  # happy path!
end
```

I think this would cover the common error case. We can add different error cases later if people run into things often. Thoughts?